ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivacyController.java

```diff
@@ -173,4 +187,49 @@ public long getSenderNonce(final Address sender, final String privacyGroupId) {
             // private state does not exist
             Account.DEFAULT_NONCE);
   }
+
+  private SendRequest createSendRequest(final PrivateTransaction privateTransaction) {
+    final BytesValueRLPOutput bvrlp = new BytesValueRLPOutput();
+    privateTransaction.writeTo(bvrlp);
+    final String payload = BytesValues.asBase64String(bvrlp.encoded());
+
+    if (privateTransaction.getPrivacyGroupId().isPresent()) {
+      return new SendRequestBesu(
+          payload,
+          enclavePublicKey,
+          BytesValues.asBase64String(privateTransaction.getPrivacyGroupId().get()));
+    } else {
+      final List<String> privateFor =
+          privateTransaction.getPrivateFor().get().stream()
+              .map(BytesValues::asBase64String)
+              .collect(Collectors.toList());
+
+      // FIXME: orion should accept empty privateFor
+      if (privateFor.isEmpty()) {
+        privateFor.add(BytesValues.asBase64String(privateTransaction.getPrivateFrom()));
+      }
+
+      return new SendRequestLegacy(
+          payload, BytesValues.asBase64String(privateTransaction.getPrivateFrom()), privateFor);
+    }
+  }
+
+  private String getPrivacyGroup(final String key, final PrivateTransaction privateTransaction) {
```

Comment:
nit: Wondering if this should be split up - the function has a bit of a mismatch of meanings - i.e. why both a PrivTxn AND a Key - when they're really an and/or.
Maybe the http stuff should break out to a separate function, and the "if/else" check performed outside of this function?